app/views/ladder/LadderView.coffee

```diff
@@ -70,9 +71,12 @@ module.exports = class LadderView extends RootView
       else
         @listenToOnce @league, 'sync', @onCourseInstanceLoaded
 
-  onCourseInstanceLoaded: (@courseInstance) ->
+  onCourseInstanceLoaded: co.wrap (@courseInstance) ->
     return if @destroyed
     @classroomID = @courseInstance.get('classroomID')
+    @ownerID = @courseInstance.get('ownerID')
+    @isSchoolAdmin = yield me.isSchoolAdminOf({ classroomId: @classroomID })
+    @isTeacher = yield me.isTeacherOf({ classroomId: @classroomID })
```

Comment:
Naming suggestion: `isSchoolAdminOfClass`, `isTeacherOfClass`

Reply:
@shubhi1092 I like that better, but hesitate slightly because we can pass in course instances, and the school admin administrates teachers and not classrooms. If that does not make the naming confusing, I'm all for changing it.

Reply:
This looks confusing to me because this is similar to `user.isTeacher()` and `user.isSchoolAdmin()` which checks if the user has necessary permissions.
Here we are checking if the user can access a particular teacher/classroom/course instance, so I believe there is room for a better naming here. Otherwise looks good.